jquery.flot.absRelTime.js

```diff
@@ -113,7 +150,139 @@ the the second one the date in gregorian date format.
+            var showYear = formatString.includes("yy") || formatString.includes("#d");
+            if (!showYear && !formatString.includes("MM")) {
+                return "";
+            }
+
+            // System format
+            if (formatString.includes("#d")) {
+                return Intl.DateTimeFormat(locale, {year: 'numeric', month: 'numeric', day: 'numeric'}).format(date);
+            }
+
+            // Custom format
+            var formatParts = Intl.DateTimeFormat(locale, formatOptions).formatToParts(date);
+            var formatPartsTypeList = formatParts.map(({type, value}) => { return type; });
+            var dayIndex = formatPartsTypeList.indexOf('day');
+            var monthIndex = formatPartsTypeList.indexOf('month');
+            var dayMonthDelimiter = formatParts[(monthIndex + dayIndex) / 2].value;
+            var dayValue = leftPad(formatParts[dayIndex].value);
+            var monthValue = leftPad(formatParts[monthIndex].value);
+            var yearValue = showYear ? formatParts[formatPartsTypeList.indexOf('year')].value : "";
+
+            if (showYear) {
+                var padAmount = formatString.includes("yyyy") ? 4 : 2;
+                yearValue = padNTimes(yearValue, "0", padAmount);
+            }
+
+            if (formatString.indexOf("dd") < formatString.indexOf("MM")) {
+                return dayValue + dayMonthDelimiter + monthValue + (showYear ? dayMonthDelimiter + yearValue : "");
+            } else {
+                return monthValue + dayMonthDelimiter + dayValue + (showYear ? dayMonthDelimiter + yearValue : "");
+            }
+        }
+
+        function getFractionalSecondsString(milliseconds, formatString, forceFractionalSeconds) {
+            var fractionalSecondsIndex = formatString.indexOf(".S");
+            var fractionalSecondsSearch = new RegExp(".S+", "g");
+            var fallbackDigitCount = forceFractionalSeconds ? 3 : 0;
+            var numberOfFractionalSeconds = fractionalSecondsIndex > 0 ? fractionalSecondsSearch.exec(formatString)[0].length - 1 : fallbackDigitCount;
+            var fractionalSecondsString = padNTimes(milliseconds, "0", 3);
+            fractionalSecondsString = padNTimes(fractionalSecondsString, "0", numberOfFractionalSeconds, true);
+            fractionalSecondsString = numberOfFractionalSeconds > 0 ? fractionalSecondsString.substring(0, numberOfFractionalSeconds) : "";
+            return fractionalSecondsString;
+        }
+
+        function getFormattedTimeString(date, formatString, formatOptions, locale) {
+            var showTime = formatString.includes("hh") || formatString.includes("HH") || formatString.includes("#T");
+            if (!showTime) {
+                return "";
+            }
+
+            var fractionalSecondsString = getFractionalSecondsString(date.getMilliseconds(), formatString);
+            var numberOfFractionalSeconds = fractionalSecondsString.length;
```

Comment:
This fractional seconds count variable is declared very far away from its actual use, and it doesn't add much over accessing the string length directly (where it is used to determine if we need to append the string).